### PR TITLE
Fix #260 issues with privacy policy

### LINKED ISSUE #260: Fix privacy policy sign up flow
Currently when the privacy policy is enabled and a user clicks signup their account is created before they accept the privacy policy. We need to change the privacy policy flow to not save a user until they have agreed to the privacy policy.

image

### PR DESCRIPTION
**Pull request by jiama843**
Changed privacy policy workflow from:

Input Credentials -> Create user -> Accept terms -> Redirect to Main Room

to:

Input Credentials and Accept terms -> Create User -> Redirect to Main Room

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -26,12 +26,16 @@
   get '/signup', to: 'users#new', as: :signup
   post '/signup', to: 'users#create', as: :create_user
 
+  # Redirect to terms page
+  match '/terms', to: 'users#terms', via: [:get, :post]
```
Comment (jfederico) [inline config/routes.rb]: Why do you have get and post?

File: config/routes.rb [hunk 2]
```diff
@@ -26,12 +26,16 @@
...
+
+  # Accept terms for existing accounts
+  get '/accept', to: 'users#accept', as: :accept
```
Comment (jfederico) [inline config/routes.rb]: Is this really necessary? Can't be just an update to the terms?

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -97,14 +97,16 @@ def destroy
...
+    # Render the terms page
+    render "terms"
+  end
 
-    if params[:accept] == "true"
-      current_user.update_attribute(accepted_terms: true)
-      redirect_to current_user.main_room
-    end
+  # GET /accept
+  def accept
```
Comment (jfederico) [inline app/controllers/users_controller.rb]: lets be more ecplicit here. Lets call the method accept_terms

File: app/controllers/rooms_controller.rb [hunk 4]
```diff
@@ -17,7 +17,6 @@
 # with BigBlueButton; if not, see <http://www.gnu.org/licenses/>.
 
 class RoomsController < ApplicationController
```
Comment (jfederico) [inline app/controllers/rooms_controller.rb]: Code for validation in the room has to be resored

File: config/routes.rb (hunk 1 above)
Comment (jiama843) [reply config/routes.rb]: Yeah, post is used to accept the terms

**Pull request merged**